subsys/bluetooth/mesh/settings.c

```diff
@@ -29,6 +29,11 @@
 #include "settings.h"
 #include "cfg.h"
 
+#ifndef CONFIG_BT_MESH_CONTINUOUS_STORE
+#define CONFIG_BT_MESH_STORE_TIMEOUT 0
+#define CONFIG_BT_MESH_RPL_STORE_TIMEOUT 0
+#endif
```

Comment:
I see this convention has already crept into some parts of the tree, but I don't think it's a good idea to go synthesizing defines that look like they originate from Kconfig even though they don't actually originate from Kconfig. This can lead to some debugging frustration when you look through the Kconfig definition and can't make sense of why some macro has a value that it has. Instead, I think it would make more sense to use new names in this case, so something like:
```
#ifdef CONFIG_BT_MESH_CONTINUOUS_STORE
#define MESH_STORE_TIMEOUT CONFIG_BT_MESH_STORE_TIMEOUT
#define MESH_RPL_STORE_TIMEOUT CONFIG_BT_MESH_STORE_TIMEOUT
#else
#define MESH_STORE_TIMEOUT 0
#define MESH_RPL_STORE_TIMEOUT 0
#endif
```

Reply:
Of course the even better solution is to try to ensure that the Kconfig defines aren't attempted to be used in places where they're not relevant. Maybe that's where you're missing #ifdefs?

Reply:
Thanks! I would just go with the first option as it allows compiles the code. The parts of the code that shouldn't be included are wrapped into `IS_ENABLED`.